### ISSUE TITLE
[Feature Request] Ability to only download an arbitrary section of a video

### ISSUE THREAD

**Issue by marsmathis** [JunkFood02/Seal#516]
### Checklist

- [ ] This feature I'm requesting is already implemented in yt-dlp.
- [ ] This feature is merely a UI/UX update.
- [X] This feature is suitable for primary users with little knowledge about yt-dlp.
- [X] This feature is available for most websites, not only the video platform I use.
- [X] This feature is suitable for a large variety of videos.
- [X] This feature is not going to conflict with many of the existing options.

### Is your feature request related to a problem? Please describe.

I want to be able to only download an arbitrary section of a video (or the audio portion of that section).

### Describe the solution you'd like

I'd like to see an optional step in the download process (maybe to be enabled in settings) where once you inserted the link of the video, a timeline of the video itself pops up with a beginning and ending marker which you can move. You can hit a play button to play your currently selected section. There should be buttons to increase/decrease both markers by 1 s/0.1 s/1 frame in each direction. Once you're happy with the selection, you can go forward and select if you just want the audio or also video, just like Seal does it now. 

### Additional context

This should improve usability for any video where people might wanna cut out just a section. Right now my workflow is to download the entire video (a lot of overhead if the video is, say, an hour long but I only need a few seconds for a soundboard or whatever) and then cut it with a separate app. This could be streamlined in my opinion.

**Comment by BobbyESP** [comment]
But Seal is just a GUI for yt-dlp so I don't think it will be a soon feature/possible implementation.

Wait until Junkfood's answer

**Comment by JunkFood02** [comment]
https://reddit.com/r/youtubedl/w/howdoidownloadpartsofavideo

TL;DR: this app is not a video editor, and the video trimming can only happen after when the complete video is downloaded
